code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -101,8 +101,9 @@
 		if(!bcell)
 			return
 
+		bcell.loc = get_turf(src)
```

Comment:
Don't manipulate `loc` directly. That's what `forceMove()` is for.

Reply:
this is what the reloading mechanic does for weapons.